File: ios/Debugger/GimbalDebuggerModule.m
#import "GimbalDebuggerModule.h"
#import <React/RCTConvert.h>

@implementation GimbalDebuggerModule

RCT_EXPORT_MODULE(GimbalDebugger)

#pragma mark - Beacon Sightings Logging

RCT_EXPORT_METHOD(enableBeaconSightingsLogging)
{
    [GMBLDebugger enableBeaconSightingsLogging];
}

RCT_EXPORT_METHOD(disableBeaconSightingsLogging)
{
    [GMBLDebugger disableBeaconSightingsLogging];
}

RCT_REMAP_METHOD(isBeaconSightingsLoggingEnabled, isBeaconSightingsLoggingEnabledWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
{
  BOOL isBeaconSightingsLoggingEnabled = [GMBLDebugger isBeaconSightingsEnabled];
  resolve(@(isBeaconSightingsLoggingEnabled));
}

#pragma mark - Debug Logging

RCT_EXPORT_METHOD(enableDebugLogging)
{
    [GMBLDebugger enableDebugLogging];
}

RCT_EXPORT_METHOD(disableDebugLogging)
{
    [GMBLDebugger disableDebugLogging];
}

RCT_REMAP_METHOD(isDebugLoggingEnabled, isDebugLoggingEnabledWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
{
  BOOL isDebugLoggingEnabled = [GMBLDebugger isDebugEnabled];
  resolve(@(isDebugLoggingEnabled));
}

#pragma mark - Place Logging

RCT_EXPORT_METHOD(enablePlaceLogging) {
    [GMBLDebugger enablePlaceLogging];
}

RCT_EXPORT_METHOD(disablePlaceLogging) {
    [GMBLDebugger disablePlaceLogging];
}

RCT_REMAP_METHOD(isPlaceLoggingEnabled, isPlaceLoggingEnabledWithResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
{
  BOOL isPlaceLoggingEnabled = [GMBLDebugger isPlaceLoggingEnabled];
  resolve(@(isPlaceLoggingEnabled));
}

@end

File: android/src/main/java/com/gimbal/react/GimbalDebuggerModule.java
package com.gimbal.react;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReactApplicationContext;

import com.gimbal.android.GimbalDebugger;

import javax.annotation.Nonnull;

public class GimbalDebuggerModule extends ReactContextBaseJavaModule {

  public GimbalDebuggerModule(@Nonnull ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Nonnull
  @Override
  public String getName() {
      return "GimbalDebugger";
  }

  @ReactMethod
  public void disableBeaconSightingsLogging() {
    GimbalDebugger.disableBeaconSightingsLogging();
  }

  @ReactMethod
  public void disablePlaceLogging() {
    GimbalDebugger.disablePlaceLogging();
  }

  @ReactMethod
  public void disableDebugLogging() {
    GimbalDebugger.disableStatusLogging();
  }

  @ReactMethod
  public void enableBeaconSightingsLogging() {
    GimbalDebugger.enableBeaconSightingsLogging();
  }

  @ReactMethod
  public void enablePlaceLogging() {
    GimbalDebugger.enablePlaceLogging();
  }

  @ReactMethod
  public void enableDebugLogging() {
    GimbalDebugger.enableStatusLogging();
  }

  @ReactMethod
  public void isBeaconSightingsLoggingEnabled(Promise promise) {
    promise.resolve(GimbalDebugger.isBeaconSightingsLoggingEnabled());
  }

  @ReactMethod
  public void isPlaceLoggingEnabled(Promise promise) {
    promise.resolve(GimbalDebugger.isPlaceLoggingEnabled());
  }

  @ReactMethod
  public void isDebugLoggingEnabled(Promise promise) {
    promise.resolve(GimbalDebugger.isStatusLoggingEnabled());
  }
}
